### PR TITLE
59 fix sign up bug

### PR DESCRIPTION
**Pull request by danielKNS**

### CODE REVIEW

File: src/SignUp.tsx [hunk 1]
```diff
@@ -68,20 +57,41 @@ const SignUp = () => {
...
+					lastName,
+					email,
+					password,
+					confirmPassword,
+				})
+				.then(() => {
+					setSuccessful("Your sign-up as been succesfull. Please log-in");
+					setTimeout(() => {
+						navigate("/login");
+					}, 2500);
```
Comment (LucasCostaAtCyberSaint) [inline src/SignUp.tsx]: we don't need to wait 2.5 secs to route the user to the /login

File: src/SignUp.tsx [hunk 2]
```diff
@@ -68,20 +57,41 @@ const SignUp = () => {
...
+			axios
+				.post("http://localhost:8080/customers", {
+					firstName,
+					lastName,
+					email,
+					password,
+					confirmPassword,
+				})
+				.then(() => {
+					setSuccessful("Your sign-up as been succesfull. Please log-in");
```
Comment (LucasCostaAtCyberSaint) [inline src/SignUp.tsx]: you probably will not need this then

File: src/SignUp.tsx [hunk 3]
```diff
@@ -68,20 +57,41 @@ const SignUp = () => {
...
+					setTimeout(() => {
+						navigate("/login");
+					}, 2500);
+				})
+				.catch(errors => {
+					if (errors.response.data.error === "Bad Request") {
+						setError(errors.response.data.message);
+					}
+				});
```
Comment (LucasCostaAtCyberSaint) [inline src/SignUp.tsx]: it does not look like we're showing the proper error from the backend

File: src/SignUp.tsx (hunk 1 above)
Comment (danielKNS) [reply src/SignUp.tsx]: You are right, i will remove it

File: src/SignUp.tsx (hunk 2 above)
Comment (danielKNS) [reply src/SignUp.tsx]: Already removed

File: src/SignUp.tsx (hunk 3 above)
Comment (danielKNS) [reply src/SignUp.tsx]: It will be fixed in the backend.

**Review by aMotherFox — DISMISSED** [review]

File: src/SignUp.tsx [hunk 4]
```diff
@@ -68,13 +56,30 @@ const SignUp = () => {
...
+					confirmPassword,
+				})
+				.then(() => {
+					navigate("/login");
+				})
+				.catch(errors => {
+					if (errors.response.data.error === "Bad Request") {
+						setError(errors.response.data.message);
+					}
+				});
```
Comment (LucasCostaAtCyberSaint) [inline src/SignUp.tsx]: can we check against a 400 code instead?

Comment (danielKNS) [reply src/SignUp.tsx]: Sure

**Review by LucasCostaAtCyberSaint — APPROVED** [review]

**Review by aMotherFox — APPROVED** [review]

**Review by vportela — APPROVED** [review]

**Pull request merged**